src/pages/workspace/distanceRates/PolicyDistanceRateDetailsPage.tsx

```diff
@@ -38,22 +40,29 @@ function PolicyDistanceRateDetailsPage({route}: PolicyDistanceRateDetailsPagePro
     const policyID = route.params.policyID;
     const [policy] = useOnyx(`${ONYXKEYS.COLLECTION.POLICY}${route.params.policyID}`, {canBeMissing: true});
     const rateID = route.params.rateID;
-    const customUnit = getDistanceRateCustomUnit(policy);
+    const customUnit = useMemo(() => getDistanceRateCustomUnit(policy), [policy]);
     const rate = customUnit?.rates[rateID];
     const customUnitID = customUnit?.customUnitID;
-    const [policyReports] = useOnyx(ONYXKEYS.COLLECTION.REPORT, {
-        selector: (reports) => {
+
+    const policyReportsSelector = useCallback(
```

Comment:
@TMisiukiewicz CAn you please note down to take these out once the refactoring of this one will be underway? thanks!